### ISSUE TITLE
IgnoredReturnValue false negative case

### ISSUE THREAD

**Issue by renannprado** [detekt/detekt#5806]
## Expected Behavior

It could be that I have different expectations towards this rule, but here it goes.

I would expect IgnoredReturnValue to be raised if there's a function call whose return value is ignored.

## Observed Behavior

No error or warning raised.

## Steps to Reproduce

Given this configuration:

```yaml
potential-bugs:
  IgnoredReturnValue:
    active: true
    restrictToConfig: false
    returnValueTypes:
      - "*"
```

or this

```yaml
potential-bugs:
  IgnoredReturnValue:
    active: true
    restrictToConfig: false
    returnValueTypes:
      - "kotlin.String"
      - "java.lang.String"
```

and this piece of code:

```kotlin
fun main() {
    ignoredReturn()
}

fun ignoredReturn(): String {
    if (1 == 1) {
        return "asd"
    } else {
        return "dsa"
    }
}
```

or this piece of code:

```kotlin
fun main() {
    "abc".toIntOrNull() 
}
```

I would expect that there's a detection of `IgnoredReturnValue` at the `ignoredReturn` function call.

## Context

I'm trying to ensure that, throughout the codebase, we don't call functions whose return values are ignored in general, not only for specific situations.

Ignored return values often result in bugs and if I have to annotate all functions, that might be forgotten. Also, I can't annotate functions that are not part of my own codebase.

## Your Environment

* Version of detekt used: 1.22.0
* Version of Gradle used (if applicable): 7.6
* Gradle scan link (add `--scan` option when running the gradle task): If this is a must to be able to solve this issue, then I'll provide it
* Operating System and version: MacOS Ventura 13.1
* Link to your project (if it's a public repository): It's not public

**Comment by cortinico** [comment]
I wasn't able to reproduce this locally. I've also added a test for your scenarios here:
- https://github.com/detekt/detekt/pull/5834

```
    returnValueTypes:
      - "*"
```

Side note: this is definitely wrong (i.e. we don't support this notation).

For what you want to achieve, this config should be sufficient:

```
potential-bugs:
  active: true
  IgnoredReturnValue:
    active: true
    restrictToConfig: false
```

**Comment by renannprado** [comment]
@cortinico I'm not sure about the type resolution. I'll give it a try and let you know.
If you have a small "reproducer", could you please zip it and attach here? Might be I'm missing something.

**Comment by cortinico** [comment]
The reproducer is here: https://github.com/detekt/detekt/pull/5834
If you're looking for a project, it's easier for me if you create it and I look through it (sorry we're just too few and we can't also create reproducers for every issue we receive :'( )

**Comment by renannprado** [comment]
@cortinico no worries. Once I give a try, I'll reply here again.

Thanks!
